### PR TITLE
New WebUI integration tests Storage encryption and Wizard navigation

### PR DESCRIPTION
**Pull request by velezd**
These will serve as examples for new integration tests and should be run as part of periodic testing. I also added some missing log_step decorators to storage helper functions and new post_install_step function to IntegrationTest class.

### CODE REVIEW

File: ui/webui/test/integration/storage_encryption.tc.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  type: anaconda-webui
+  automation_data:
+    script_file: ./ui/webui/test/integration/storage_encryption.py
+    test_case: StorageEncryption
+instructions:
+  setup:
+    - Start installation in VM
+  steps:
+    - step: Go through the required steps
+    - step: On Disk e ncryption step select yes
```
Comment (rvykydal) [inline ui/webui/test/integration/storage_encryption.tc.yaml:18]: A typo in "e ncryption"

**Review by rvykydal — APPROVED** [review]
Looks great to me, thank you!

File: ui/webui/test/integration/storage_encryption.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            if [ -x /sbin/zipl ]; then
+                /sbin/zipl
+            fi
+        ''')
+        self.machine.execute('chroot /mnt/sysroot bash /root/add_keyfile.sh')
+
+    @log_step(docstring=True)
+    def check_installed_system(self):
+        """ Check encryption is enabled """
+        self.assertTrue('is active and is in use' in self.machine.execute('cryptsetup status /dev/mapper/luks*'))
```
[the remark this answers is not in the document]
Comment (velezd) [reply ui/webui/test/integration/storage_encryption.py:57]: Done, anything the test prints will end-up in the test output log

File: ui/webui/test/integration/storage_encryption.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            chmod 0600 /etc/crypttab
+            kernel_file=`grubby --default-kernel`
+            kernel_version=`rpm -qf $kernel_file --qf '%{VERSION}-%{RELEASE}.%{ARCH}'`
+            initrd_file="/boot/initramfs-${kernel_version}.img"
+            dracut -f -I /root/keyfile $initrd_file $kernel_version
+            if [ -x /sbin/zipl ]; then
+                /sbin/zipl
+            fi
+        ''')
+        self.machine.execute('chroot /mnt/sysroot bash /root/add_keyfile.sh')
```
[the remark this answers is not in the document]
Comment (velezd) [reply ui/webui/test/integration/storage_encryption.py]: Done

File: ui/webui/test/integration/storage_encryption.tc.yaml (hunk 1 above)
Comment (velezd) [reply ui/webui/test/integration/storage_encryption.tc.yaml]: Fixed, thanks

File: ui/webui/test/integration/storage_encryption.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+import os
+import sys
+
+# import Cockpit's machinery for test VMs and its browser test API
+TEST_DIR = os.environ['WEBUI_TEST_DIR']
+sys.path.append(TEST_DIR)
+sys.path.append(os.path.join(TEST_DIR, "common"))
+sys.path.append(os.path.join(TEST_DIR, "helpers"))
+sys.path.append(os.path.join(os.path.dirname(TEST_DIR), "bots/machine"))
```
[the remark this answers is not in the document]
Comment (velezd) [reply ui/webui/test/integration/storage_encryption.py:26]: This is inherited from unit tests, I agree it should be improved in the future, but that is out of scope for this PR.

[the remark this answers is not in the document]
Comment (VladimirSlavik) [reply ui/webui/test/integration/storage_encryption.py:26]: Maybe reach out to the cockpit folks helping us? IIRC they did something about it.

File: ui/webui/test/helpers/storage.py [hunk 5]
```diff
@@ -176,3 +186,21 @@ def check_pw_strength(self, strength):
...
+            chmod 0600 /etc/crypttab
+            kernel_file=`grubby --default-kernel`
+            kernel_version=`rpm -qf $kernel_file --qf '%{VERSION}-%{RELEASE}.%{ARCH}'`
+            initrd_file="/boot/initramfs-${kernel_version}.img"
+            dracut -f -I /root/keyfile $initrd_file $kernel_version
+            if [ -x /sbin/zipl ]; then
+                /sbin/zipl
+            fi
+        ''')
+        self.machine.execute('chroot /mnt/sysroot bash /root/add_keyfile.sh')
```
Comment (VladimirSlavik) [inline ui/webui/test/helpers/storage.py:206]: Github says there's no EOL here...

**Review by VladimirSlavik — APPROVED** [review]
Looks good to me. Thank you!

**Review by jstodola — APPROVED** [review]
Thanks

File: ui/webui/test/helpers/storage.py (hunk 5 above)
Comment (velezd) [reply ui/webui/test/helpers/storage.py]: Fixed, thanks

**Pull request merged**